File: Student.py
class Person:
    name = None
    age = None
    hometown = None
    stays = None

    def __init__(self):
        self.__gender = None

    def addYear(self):
        age += 1

    def setGender(self, pGender):
        self.__gender = pGender

    def getGender():
        return self.__gender


class Student(Person):
    def __init__(self):
        super().__init__()
        self.__schoolName = ""
        self.__year = 0

    def setSchoolName_Year(self, sName, sYear):
        self.__schoolName = sName
        self.__year = sYear

    def getSchoolName(self):
        return self.__schoolName

    def getYear(self):
        return self.__year


class Worker(Person):
    def __init__(self):
        super().__init__()
        self.__org = ""
        self.__year = 0
        self.__salary = 0

    def setOrg_Year_Salary(self, orgName, startYear, orgSalary):
        self.__org = orgName
        self.__year = startYear
        self.__salary = orgSalary

    def getOrg(self):
        return self.__org

    def getYear(self):
        return self.__year

    def getSalary(self):
        return self.__salary


class StudentWorker(Student, Worker):
    def __init__(self, pName, pAge):
        super().__init__()
        self.name = pName
        self.age = pAge


def main():
    #Object code goes here

if __name__ == __main__:
    main()

File: Student.cpp
#include <iostream>
#include <string>

using namespace std;

class Person {
	public:
		string name;
		int age;
		string hometown;
		string stays;
		void addYear() {
			age++;
		}
		void setGender(char pGender) {
			gender = pGender;
		}
		char getGender() {
			return gender;
		}
	private:
		char gender;
};

class Student : virtual public Person {
	public:
		void setSchoolName_Year(string sName, long int sYear) {
			schoolName = sName;
			year = sYear;
		}
		string getSchoolName() {
			return schoolName;
		}
		long int getYear() {
			return year;
		}
	private:
		string schoolName;
		long int year;
};

class Worker : virtual public Person {
	public:
		void setOrg_Year_Salary(string orgName, long int startYear, int orgSalary) {
			org = orgName;
			year = startYear;
			salary = orgSalary;
		}
		string getOrg() {
			return org;
		}
		long int getYear1() {
			return year;
		}
		int getSalary(){
			return salary;
		}

	private:
		string org;
		long int year;

	protected:
		int salary;
};

class StudentWorker : public Student, public Worker {
	public:
		string name;
		int age;
		string hometown;
		string stays;
		StudentWorker(string pName, int pAge) {
			name = pName;
			age = pAge;
		}
};

int main() {
  	//Objects go here
		return 0;
}
